### PR TITLE
Add delete documentation button and claim button

### PR DESCRIPTION
**Pull request by pascalwelte**
This only includes the possibility to claim and delete from the home page and not directly from the project doc page

### CODE REVIEW

**Comment by randombenj** [comment]
Cool stuff :tada: gonna have a look at this

**Comment by randombenj** [comment]
First of all, looks great!

However two things that confuse me a bit:

1) could you make the claim token and delete button only visible on the front page as the project/version anyway has to be choosen? Basically same behaviour as uppload button.
2) When claiming the token, there is a 'token' field wich is confusing (do I hav to enter a token there), better hide this field untill the token is there.

File: web/src/pages/Claim.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<template>
+   <Layout>
+    <form novalidate class="claim-form">
+      <h1 class="claim-title">Claim Token</h1>
+
+      <md-field>
+        <label>Token</label>
+        <md-input id="token" name="token" v-model="form.token" readonly />
+      </md-field>
```
Comment (randombenj) [inline web/src/pages/Claim.vue:9]: Could you move this field below the project name and hide it until the token is there?

File: web/src/pages/Delete.vue [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        this.error = msg
+      } catch(err) {
+        // something went wrong while
+        this.showError = true
+        this.error = err
+      }
+
+      this.sending = false
+    },
+    async load_versions() {
```
Comment (randombenj) [inline web/src/pages/Delete.vue:97]:
```suggestion
    async loadVersions() {
```

File: web/src/pages/Delete.vue [hunk 3]
```diff
@@ -0,0 +1,137 @@
+<template>
+   <Layout>
+    <form novalidate @submit.prevent="validateDelete" class="delete-form">
+      <h1 class="delete-title">Delete documentation</h1>
+
+      <md-field>
+        <label>Projectname</label>
+        <md-select type="text" id="project" name="project" v-model="form.project" :disabled="sending" @md-selected="load_versions">
```
Comment (randombenj) [inline web/src/pages/Delete.vue:8]:
```suggestion
        <md-select type="text" id="project" name="project" v-model="form.project" :disabled="sending" @md-selected="loadVersions">
```

File: web/src/pages/Docs.vue [hunk 4]
```diff
@@ -20,18 +20,23 @@
...
+    <ClaimButton class="docs-claim-button" />
+    <DeleteButton class="docs-delete-button" />
   </Layout>
 </template>
 
 <script>
 import Layout from '@/components/Layout.vue'
-
 import ProjectRepository from '@/repositories/ProjectRepository'
+import ClaimButton from '@/components/ClaimButton.vue'
```
Comment (randombenj) [inline web/src/pages/Docs.vue:31]: I would not add the buttons to the docs view, this is confusing as one would expect to delete the current docs, wich is not the case.

File: web/src/repositories/ProjectRepository.js [hunk 5]
```diff
@@ -86,5 +86,33 @@ export default {
...
+   * Deletes existing project documentation
+   * @param {string} projectName Name of the project
+   * @param {string} version Name of the version
+   */
+  async delete_doc(projectName, version, token) {
+    const headers = { "Docat-Api-Key": token }
+    await fetch(`${this.baseURL}/api/${projectName}/${version}`,
+      {
+        method: 'DELETE',
+	headers: headers
```
Comment (randombenj) [inline web/src/repositories/ProjectRepository.js:114]:
```suggestion
        headers: headers
```

File: web/tests/unit/project-repository.spec.js [hunk 6]
```diff
@@ -77,4 +77,18 @@ describe('ProjectRepository', () => {
...
+  it('should delete an existing documentation', async () => {
+    mockFetchData({})
+
+    await ProjectRepository.delete_doc('awesome-project', '1.2', '1234')
+
+    expect(global.fetch).toHaveBeenCalledTimes(1)
+    expect(global.fetch).toHaveBeenCalledWith('https://do.cat/api/awesome-project/1.2',
+      {
+        'method': 'DELETE',
+	'headers': { 'Docat-Api-Key': '1234' }
```
Comment (randombenj) [inline web/tests/unit/project-repository.spec.js:90]:
```suggestion
        'headers': { 'Docat-Api-Key': '1234' }
```

File: web/src/pages/Claim.vue (hunk 1 above)
Comment (pascalwelte) [reply web/src/pages/Claim.vue:9]: @Ajapai Can you do this?

Comment (Ajapai) [reply web/src/pages/Claim.vue:9]: I'll try

**Comment by randombenj** [comment]
Closed by @Ajapai s PR

**Pull request closed without merging**